reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java

```diff
@@ -183,50 +188,140 @@ private String buildLinkString(Iterable<LinkModel> links)
             return linksString.toString();
         }
 
-        private void writeCsvRecordForProject(Map<String, CSVWriter> projectToFile, String outputFolderPath, ProjectModel projectModel, String[] line)
+        private void produceApplicationListCSV(GraphRewrite event, Map<String, CSVWriter> projectToFile, String outputFolderPath)
+        {
+
+
+
+            ApplicationReportService applicationReportService = new ApplicationReportService(event.getGraphContext());
+            TechnologyTagService techTagService = new TechnologyTagService(event.getGraphContext());
+            List<ApplicationReportModel> apps = applicationReportService.findAll();
+            List<TechnologyTagModel> masterTagList = techTagService.findAll().stream().filter(tag -> !tag.getName().equals("Decompiled Java File")).sorted((o1,o2) ->
+
+                            {
+                                String n1 = o1.getName();
+                                String n2 = o2.getName();
+                                return n1.compareToIgnoreCase(n2);
+                            }).collect(Collectors.toList());
+
+            List<String> headerFieldsList = new ArrayList<>();
+            headerFieldsList.add("App Name");
+            ArrayList<String> appNames = new ArrayList<>();
+            ArrayList<ApplicationReportModel> distinctApps = new ArrayList<>();
+
+            apps.forEach(app ->
+            {
+                if(!appNames.contains(app.getProjectModel().getRootFileModel().getFileName())) {
+                    appNames.add(app.getProjectModel().getRootFileModel().getFileName());
+                    distinctApps.add(app);
+                }
+            });
+
+
+
+            masterTagList.forEach( masterTag -> headerFieldsList.add(masterTag==null?"":masterTag.getName()));
```

Comment:
Following the code I don't see a way where masterTag could be a null

Reply:
I agree - null catch removed